modules/transcription-service-google-speech-impl/src/main/java/org/opencastproject/transcription/googlespeech/GoogleSpeechTranscriptionService.java

```diff
@@ -866,6 +870,37 @@ private String buildResultsFileName(String jobId) {
     return PathSupport.toSafeName(jobId + ".json");
   }
 
+  private void cancelTranscription(String jobId, String message) {
+    try {
+      database.updateJobControl(jobId, TranscriptionJobControl.Status.Canceled.name());
+      String mpId = database.findByJob(jobId).getMediaPackageId();
+      try {
+        // Delete file stored on Google storage
+        String token = getRefreshAccessToken();
+        deleteStorageFile(mpId, token);
+      } catch (Exception ex) {
+        logger.warn(String.format("could not delete file %s.%s from Google cloud storage", mpId, defaultEncoding), ex);
+      }
+      // Send notification email
+      sendEmail("Transcription ERROR", String.format("%s(media package %s, job id %s).", message, mpId, jobId));
+    } catch (Exception e) {
+      logger.error(String.format("ERROR while deleting transcription job: %s", jobId), e);
+    }
```

Comment:
It might be helpful to put the sendMail in a finally block to include the general Exception, or, condense into one try-catch block.

Reply:
fixed